part4/006.js

```diff
@@ -3,25 +3,26 @@
  */
 
 function solution(N, stages) {
-  const failureRate = {};
+  const failureRates = [];
 
-  let total = stages.length;
-  for (let i = 1; i <= N; i++) {
-    let count = 0;
-    for (let j = 0; j < stages.length; j++) {
-      if (i === stages[j]) count++;
-    }
-    failureRate[i] = count / total;
-    total = total - count;
-    count = 0;
-  }
+  const counts = Array.from({ length: N }).fill(0);
+  stages.forEach((currentStage) => {
+    if (currentStage - 1 >= N) return;
+    counts[currentStage - 1]++;
```

Comment:
early return 대신 조건에서만 해당 로직이 돌도록 변경 가능
(아래, 라인 22~23 도 마찬가지)

```suggestion
if (currentStage - 1  < N) { // or  if (currentStage  <= N)
    counts[currentStage - 1]++;
}
```